File: ExcelSheetColumnNumber/ExcelSheetColumnNumber.py
"""
ORD() => Возвращает числовое представление для указанного символа.
"""

class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
        res = 0
        for i, c in enumerate(reversed(columnTitle.upper())):
            res += (26 ** i) * (ord(c) - ord("A") + 1)
        return res  

File: ExcelSheetColumnNumber/ExcelSheetColumnNumber.js
// Метод charCodeAt() возвращает числовое значение Юникода для символа по указанному индексу

/**
 * @param {string} columnTitle
 * @return {number}
 */
var titleToNumber = function (columnTitle) {
  let res = 0;
  for (let i = 0; i < columnTitle.length; i++) {
    res = res * 26 + columnTitle.charCodeAt(i) - "A".charCodeAt(0) + 1;
  }
  return res;
};
